File: ui/frontend-react/src/types/evidence.ts
/**
 * TypeScript types for Evidence Search & Filter UI.
 *
 * Story 6-9: Mirrors backend Pydantic schemas for type-safe API communication.
 */

/**
 * Evidence list item (compact for grid view).
 */
export interface Evidence {
  id: string;
  competitor_name: string;
  violation_type: string;
  severity: string;
  claim_text: string;
  claim_category: string;
  source_type: string;
  source_url: string;
  captured_at: string;
  screenshot_path: string;
  screenshot_hash: string;
}

/**
 * Audit log entry for evidence operations.
 */
export interface AuditLogEntry {
  id: string;
  action: string;
  actor: string;
  details: Record<string, unknown> | null;
  hash_verified: boolean | null;
  created_at: string;
}

/**
 * Full evidence detail with relations.
 */
export interface EvidenceDetail {
  id: string;
  competitor_name: string;
  violation_type: string;
  severity: string;
  claim_text: string;
  claim_category: string;
  source_type: string;
  source_url: string;
  captured_at: string;
  screenshot_path: string;
  screenshot_hash: string;
  screenshot_size_bytes: number;
  regulation_violated: string | null;
  detection_reasoning: string | null;
  confidence: number;
  evidence_metadata: Record<string, unknown> | null;
  created_at: string;
  audit_logs: AuditLogEntry[];
}

/**
 * Paginated evidence list response.
 */
export interface EvidenceListResponse {
  items: Evidence[];
  total_count: number;
  has_more: boolean;
}

/**
 * Evidence summary statistics.
 */
export interface EvidenceSummary {
  total_evidence: number;
  by_severity: Record<string, number>;
  by_competitor: Record<string, number>;
  by_violation_type: Record<string, number>;
}

/**
 * Filter parameters for evidence search.
 */
export interface EvidenceFilters {
  competitor?: string;
  violationType?: string;
  severity?: string;
  dateFrom?: string;
  dateTo?: string;
  keywords?: string;
  sourceType?: string;
}

/**
 * Monthly violation count for competitor trend chart.
 */
export interface CompetitorTimelineEntry {
  month: string;
  count: number;
}

/**
 * Severity level display colors (Tailwind classes).
 */
export const SEVERITY_COLORS: Record<string, string> = {
  high: "bg-red-100 text-red-800",
  medium: "bg-yellow-100 text-yellow-800",
  low: "bg-green-100 text-green-800",
};

/**
 * Violation type display labels.
 */
export const VIOLATION_TYPE_LABELS: Record<string, string> = {
  unauthorized_treatment_claim: "Treatment Claim",
  unauthorized_prevention_claim: "Prevention Claim",
  unauthorized_enhancement_claim: "Enhancement Claim",
  unauthorized_wellness_claim: "Wellness Claim",
  violation: "Violation",
  suspect: "Suspect",
};

/**
 * Source type display labels.
 */
export const SOURCE_TYPE_LABELS: Record<string, string> = {
  instagram_post: "Instagram",
  website_page: "Website",
};

File: ui/backend/schemas/evidence.py
"""Pydantic schemas for Evidence API.

Story 6-9: Searchable Evidence Database.
Provides response schemas for evidence search, detail, and summary endpoints.

Schemas:
    - EvidenceListItemSchema: Evidence summary for list view
    - EvidenceDetailSchema: Full evidence detail with relations
    - EvidenceListResponse: Paginated evidence list
    - EvidenceSummarySchema: Aggregate counts by severity, competitor, violation_type
    - CompetitorTimelineEntry: Monthly violation count for trend chart
    - AuditLogEntrySchema: Single audit log record
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class EvidenceListItemSchema(BaseModel):
    """Evidence summary for list view.

    Compact representation for evidence grid display.
    """

    id: str = Field(..., description="Evidence UUID")
    competitor_name: str = Field(..., description="Competitor name")
    violation_type: str = Field(..., description="Violation type descriptor")
    severity: str = Field(..., description="Severity level (high/medium/low)")
    claim_text: str = Field(..., description="Claim text (may be truncated)")
    claim_category: str = Field(..., description="Claim category")
    source_type: str = Field(..., description="Source type (instagram_post/website_page)")
    source_url: str = Field(..., description="Original source URL")
    captured_at: datetime = Field(..., description="When screenshot was captured")
    screenshot_path: str = Field(..., description="Relative path to screenshot")
    screenshot_hash: str = Field(..., description="SHA-256 hash of screenshot")

    model_config = {"from_attributes": True}


class AuditLogEntrySchema(BaseModel):
    """Single audit log record."""

    id: str = Field(..., description="Audit log UUID")
    action: str = Field(..., description="Operation type")
    actor: str = Field(..., description="Who performed the action")
    details: Optional[dict] = Field(default=None, description="Operation-specific data")
    hash_verified: Optional[bool] = Field(
        default=None, description="Integrity check result"
    )
    created_at: datetime = Field(..., description="When action occurred")

    model_config = {"from_attributes": True}


class EvidenceDetailSchema(BaseModel):
    """Full evidence detail with violation and audit log relations.

    Displayed in the evidence detail drawer/panel.
    """

    id: str = Field(..., description="Evidence UUID")
    competitor_name: str = Field(..., description="Competitor name")
    violation_type: str = Field(..., description="Violation type descriptor")
    severity: str = Field(..., description="Severity level")
    claim_text: str = Field(..., description="Full claim text")
    claim_category: str = Field(..., description="Claim category")
    source_type: str = Field(..., description="Source type")
    source_url: str = Field(..., description="Original source URL")
    captured_at: datetime = Field(..., description="Screenshot capture time")
    screenshot_path: str = Field(..., description="Relative path to screenshot")
    screenshot_hash: str = Field(..., description="SHA-256 hash")
    screenshot_size_bytes: int = Field(..., description="File size in bytes")
    regulation_violated: Optional[str] = Field(
        default=None, description="EU regulation reference"
    )
    detection_reasoning: Optional[str] = Field(
        default=None, description="Classification reasoning"
    )
    confidence: float = Field(..., description="Confidence score 0.0-1.0")
    evidence_metadata: Optional[dict] = Field(
        default=None, description="Additional metadata"
    )
    created_at: datetime = Field(..., description="Record creation time")
    audit_logs: list[AuditLogEntrySchema] = Field(
        default_factory=list, description="Audit log timeline"
    )

    model_config = {"from_attributes": True}


class EvidenceListResponse(BaseModel):
    """Paginated evidence list."""

    items: list[EvidenceListItemSchema] = Field(
        ..., description="Evidence items for current page"
    )
    total_count: int = Field(..., ge=0, description="Total matching evidence")
    has_more: bool = Field(..., description="Whether more results exist")

    model_config = {"from_attributes": True}


class EvidenceSummarySchema(BaseModel):
    """Aggregate evidence statistics.

    Dashboard overview counts.
    """

    total_evidence: int = Field(..., ge=0, description="Total evidence records")
    by_severity: dict[str, int] = Field(
        ..., description="Evidence count per severity level"
    )
    by_competitor: dict[str, int] = Field(
        ..., description="Evidence count per competitor"
    )
    by_violation_type: dict[str, int] = Field(
        ..., description="Evidence count per violation type"
    )

    model_config = {"from_attributes": True}


class CompetitorTimelineEntry(BaseModel):
    """Monthly violation count for competitor trend chart."""

    month: str = Field(..., description="Month in YYYY-MM format")
    count: int = Field(..., ge=0, description="Violation count for this month")

    model_config = {"from_attributes": True}


__all__ = [
    "AuditLogEntrySchema",
    "CompetitorTimelineEntry",
    "EvidenceDetailSchema",
    "EvidenceListItemSchema",
    "EvidenceListResponse",
    "EvidenceSummarySchema",
]
